### PR TITLE
Clean up time management

### PR DESCRIPTION
**Pull request by dnoursi**
Most of this code transitions the units of time from seconds to cycles, which makes our data more consistent. The previous code created `csv`s which were difficult to graph, because intervals would be effectively randomly sampled from 19,20, and 21.

Because the periods of corpus snapshots are regulated so successfully, and margins of error in my experiments now never exceed 1 second, this is justified. In particular, because time is tracked with `$SECONDS`, an offset of 1 second isn't a useful nuance to record and graph. 

As a failsafe, seconds are still recorded, and they will be processed by the golang parser to ensure nothing crazy happens.

### CODE REVIEW

**Comment by morehouse** [comment]
What if we decide to change the time between cycles in the future?  Wouldn't this commit make it difficult to compare previous results with new?

**Comment by morehouse** [comment]
Also, what is difficult about graphing seconds?

**Comment by dnoursi** [comment]
The crux of the issue is that using seconds put each benchmark at its own pace, but most of the difference between the benchmarks was noise, e.g. one would snapshot at times
20, 41, 62, 81, while another would record at
21, 41, 61, 82

This was a somewhat minor issue, but it's significant if I'd like to combine or rearrange columns between CSVs

**Comment by dnoursi** [comment]
It is important to record `WAIT_PERIOD`; this could almost be inferred from the seconds records, which still exist. I think it would make a lot of sense to put an artifact named e.g. `wait-period.txt` next to those `seconds-${CYCLE}` files, and these together would be sufficient

**Comment by morehouse** [comment]
> The crux of the issue is that using seconds put each benchmark at its own pace, but most of the difference between the benchmarks was noise

Doesn't that just mean the data points will be translated slightly to the left or right on the graph?  And isn't that what we would want for an accurate comparison rather than pretending all snapshots happened at exactly the same times.

And I still don't understand what's more difficult about graphing seconds than cycles.  Are you trying to do a bar graph rather than a plot?  If so, why?

**Comment by dnoursi** [comment]
I am creating line plots, but it seems most supported to have a unified x-axis for multiple y-axes, for example https://gist.github.com/mbostock/3884955#file-data-tsv

I'd like to do combine CSVs in different contexts: for considering multiple trials of the same benchmark/fengine, and also for considering multiple fengines executed on the same benchmark. There may be other cases, and it feels overall less restrictive to freely combine and interchange CSV columns.

The second case complicates things somewhat further if we'd like to compute the median of all trials; if we'd like to really consider different trials as being collected some small window apart, computing any form of average would require some form of time series scaling (rudimentary calculus?) to account for the second of additional fuzzing

**Comment by dnoursi** [comment]
If preferable, I could keep only part of this PR, and revert time units to seconds: the most important was assigning 

```
  NEXT_SYNC=$(($CYCLE * $WAIT_PERIOD))
```
This added all rigidity, preventing a runway sequence of additional seconds each trial

**Comment by morehouse** [comment]
> I'd like to do combine CSVs in different contexts: for considering multiple trials of the same benchmark/fengine, and also for considering multiple fengines executed on the same benchmark. There may be other cases, and it feels overall less restrictive to freely combine and interchange CSV columns.

So this is more of a CSV difficulty than a graphing one...

> The second case complicates things somewhat further if we'd like to compute the median of all trials; if we'd like to really consider different trials as being collected some small window apart, computing any form of average would require some form of time series scaling (rudimentary calculus?) to account for the second of additional fuzzing

I see.  Well if times are only +-1, its simpler to just do what this PR is proposing.

File: engine-comparison/dispatcher.sh [hunk 1]
```diff
@@ -217,14 +223,14 @@ measure_coverage () {
...
-  echo "$VM_SECONDS, $($WORK/coverage-builds/sancov.py print $SANCOV_DIR/* | wc -w)" >> $REPORT_DIR/coverage-graph.csv
-  echo "$VM_SECONDS, $(wc -c $(find $CORPUS_DIR -maxdepth 1 -type f) | tail --lines=1 | grep -o [0-9]* )" >> $REPORT_DIR/corpus-size-graph.csv
-  echo "$VM_SECONDS, $(find $CORPUS_DIR -maxdepth 1 -type f | wc -l)" >> $REPORT_DIR/corpus-elems-graph.csv
+  echo "$THIS_CYCLE,$($WORK/coverage-builds/sancov.py print $SANCOV_DIR/* | wc -w)" >> $REPORT_DIR/coverage-graph.csv
+  echo "$THIS_CYCLE,$(wc -c $(find $CORPUS_DIR -maxdepth 1 -type f) | tail --lines=1 | grep -o [0-9]* )" >> $REPORT_DIR/corpus-size-graph.csv
+  echo "$THIS_CYCLE,$(find $CORPUS_DIR -maxdepth 1 -type f | wc -l)" >> $REPORT_DIR/corpus-elems-graph.csv
 
   echo "LATEST_REPORT=$THIS_CYCLE" > $REPORT_DIR/latest-report
-  gsutil -m rsync -rd $REPORT_DIR ${GSUTIL_BUCKET}/reports/${BENCHMARK}
+  gsutil -m rsync -rd $REPORT_DIR ${GSUTIL_BUCKET}/reports/${BENCHMARK}/${FENGINE_NAME}
```
Comment (morehouse) [inline engine-comparison/dispatcher.sh]: `${THIS_BENCHMARK}`

File: engine-comparison/runner.sh [hunk 2]
```diff
@@ -50,37 +49,44 @@ mkdir corpus-archives
...
-# This works for libfuzzer, tbd for AFL:
-while [[ ! -f crash* ]]; do
-# while [[ "infinite loop" ]]; do
+# Setting SECONDS is fine, it still gets ++ on schedule
+SECONDS=0
+NEXT_SYNC=$WAIT_PERIOD
+# This doesn't work for leaks or timeouts, as they have different names:
+# Currently only the individual copies of test-libfuzzer.sh document
+# the particulars of how to identify complete benchmarks
+while [[ ! $(ls | grep crash) ]]; do
```
Comment (morehouse) [inline engine-comparison/runner.sh]: A possible full solution is to check whether the `EXEC_CMD` process is still running.

File: engine-comparison/dispatcher.sh [hunk 3]
```diff
@@ -217,14 +223,14 @@ measure_coverage () {
   # Could include calls to external scripts in Golang, HTML, etc
 
   # declare VM_SECONDS
-  . $EXPERIMENT_DIR/results/seconds-${THIS_CYCLE}
+  # . $EXPERIMENT_DIR/results/seconds-${THIS_CYCLE}
```
Comment (morehouse) [inline engine-comparison/dispatcher.sh]: Delete unused code.

File: engine-comparison/dispatcher.sh [hunk 4]
```diff
@@ -134,7 +134,9 @@ elif [[ $BMARKS == 'small' ]]; then
   BENCHMARKS="c-ares-CVE-2016-5180 re2-2014-12-09"
 elif [[ $BMARKS == 'none' ]]; then
   BENCHMARKS=""
-  #elif [[ $BMARKS == 'other alias' ]]; do
+elif [[ $BMARKS == 'three' ]]; then
+  BENCHMARKS="guetzli-2017-3-30 libssh-2017-1272 re2-2014-12-09"
+#elif [[ $BMARKS == 'other alias' ]]; then
```
Comment (morehouse) [inline engine-comparison/dispatcher.sh]: Unnecessary comment.  It's pretty clear how this if-else statement works.

File: engine-comparison/runner.sh [hunk 5]
```diff
@@ -50,37 +49,44 @@ mkdir corpus-archives
...
 
   # Ensure that measurements happen every $WAIT_PERIOD
   SLEEP_TIME=$(($NEXT_SYNC - $SECONDS))
+  # if $NEXT_SYNC
```
Comment (morehouse) [inline engine-comparison/runner.sh]: Delete unused code.

**Comment by dnoursi** [comment]
> So this is more of a CSV difficulty than a graphing one...

I'd say that's accurate, my goal was to have a more uniform x-axis

File: engine-comparison/runner.sh (hunk 2 above)
Comment (dnoursi) [reply engine-comparison/runner.sh]: This is a good idea, thanks!

**Review by morehouse — APPROVED** [review]

**Pull request merged**